pkg/agent/route/route_windows.go

```diff
@@ -122,19 +129,19 @@ func (c *Client) initServiceIPRoutes() error {
 
 // Reconcile removes the orphaned routes and related configuration based on the desired podCIDRs and Service IPs. Only
 // the route entries on the host gateway interface are stored in the cache.
-func (c *Client) Reconcile(podCIDRs []string, svcIPs map[string]bool) error {
+func (c *Client) Reconcile(podCIDRs []string) error {
 	desiredPodCIDRs := sets.NewString(podCIDRs...)
-	routes, err := c.listRoutes()
+	routes, err := c.listIPRoutesOnGW()
 	if err != nil {
 		return err
 	}
 	for dst, rt := range routes {
 		if desiredPodCIDRs.Has(dst) {
-			c.hostRoutes.Store(dst, rt)
+			c.nodeRoutes.Store(dst, rt)
 			continue
 		}
-		if _, ok := svcIPs[dst]; ok {
-			c.hostRoutes.Store(dst, rt)
+		// Don't delete the routes which are added by AntreaProxy when proxyAll is enabled.
+		if c.proxyAll && c.isServiceRoute(rt) {
```

Comment:
Didn't find a function to delete stale Service routes when AntreaProxyAll is enabled. Although `AddClusterIPRoute` tries to merge the Service route entries, and removes the entries containing the svceIP added separately, what if the ClusterIP Service is removed during Agent down time. There is no mechnism to remove the routing entries for a non-existing Service, right?

Reply:
```go
if curClusterIPCIDR != nil {
		route.DestinationSubnet = curClusterIPCIDR
		staleRoutes = []*util.Route{route}
                //  This is used to get the stale merged route, and it is not for the first Service.
	} else {
		routes, err := c.listIPRoutesOnGW()
		if err != nil {
			return fmt.Errorf("error listing ip routes: %w", err)
		}
		for _, rt := range routes {
			if rt.GatewayAddress.Equal(gw) && !rt.DestinationSubnet.IP.Equal(svcIP) && rt.DestinationSubnet.Contains(svcIP) {
                                 // This is used to get stale merged routed installed before this round of AntreaProxy, and it is only for the first Service. !rt.DestinationSubnet.IP.Equal(svcIP) is used to exclude the route installed for the first Service.
				staleRoutes = append(staleRoutes, rt)
			}
		}
	}
```